gef.py

```diff
@@ -5448,35 +5491,12 @@ def do_invoke(self, argv: List) -> None:
             self.usage()
             return
 
-        modname, structname = self.get_modulename_structname_from_arg(argv[0])
-        self.__dump_structure(modname, structname)
-        return
-
-    def __dump_structure(self, mod_name: str, struct_name: str) -> None:
-        # If it's a builtin or defined in the ELF use gdb's `ptype`
-        try:
-            gdb.execute(f"ptype struct {struct_name}")
-            return
-        except gdb.error:
-            pass
-
-        self.__dump_custom_structure(mod_name, struct_name)
-        return
-
-    def __dump_custom_structure(self, mod_name: str, struct_name: str) -> None:
-        if not self.is_valid_struct(mod_name):
-            err(f"Invalid structure name '{struct_name}'")
-            return
-
-        _class, _struct = self.get_structure_class(mod_name, struct_name)
-
-        for _name, _type in _struct._fields_:
-            _size = ctypes.sizeof(_type)
-            __name = Color.colorify(_name, gef.config["pcustom.structure_name"])
-            __type = Color.colorify(_type.__name__, gef.config["pcustom.structure_type"])
-            __size = Color.colorify(hex(_size), gef.config["pcustom.structure_size"])
-            __offset = Color.boldify(f"{getattr(_class, _name).offset:04x}")
-            gef_print(f"{__offset}   {__name:32s}   {__type:16s}  /* size={__size} */")
+        _, structname = self.explode_type(argv[0])
+        manager = ExternalStructureManager()
```

Comment:
Why do we need to instantiate this? It's not really used as a class, it's just a container. If we had more than one file in gef it wouldn't even be a class, but rather a module, unless I'm missing something.

Reply:
No it's needed, forgot to add the cache part. For now, that fixes the compatibility so that all commands are back to working like before. Later on, it will be a global part of `gef.session`, which we'll be able to reset on demand.